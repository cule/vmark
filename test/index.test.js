const vmark = require('../')

function snapshot(title, text, options) {
  test(title, () => {
    expect(vmark(text.trim(), options)).toMatchSnapshot()
  })
}

snapshot('basic', `
# hello

<counter :start="0" />
`)

snapshot('with tags', `
<div class="foo">{{ foo }}</div>

<style scoped>
.foo {
  color: red;
}
</style>
`)

snapshot('wrap html', `
# hello
`, {
  wrapHTML: html => `<foo>${html}</foo>`
})

snapshot('insert code above', `
# hello

\`\`\`vue {insert: 'above'}
<button>Button</button>
\`\`\`
`)

snapshot('insert code below', `
# hello

\`\`\`vue {insert: 'below'}
<button>Button</button>
\`\`\`
`)
