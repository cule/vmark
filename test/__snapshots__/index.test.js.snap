// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic 1`] = `
"<template>
  <div class=\\"vmark\\"><h1>hello</h1>
<counter :start=\\"0\\" /></div>
</template>

"
`;

exports[`insert code above 1`] = `
"<template>
  <div class=\\"vmark\\"><h1>hello</h1>
<button>Button</button>
<pre><code class=\\"language-vue\\">&lt;button&gt;Button&lt;/button&gt;
</code></pre>
</div>
</template>

"
`;

exports[`insert code below 1`] = `
"<template>
  <div class=\\"vmark\\"><h1>hello</h1>
<pre><code class=\\"language-vue\\">&lt;button&gt;Button&lt;/button&gt;
</code></pre>
<button>Button</button>
</div>
</template>

"
`;

exports[`with tags 1`] = `
"<template>
  <div class=\\"vmark\\"><div class=\\"foo\\">{{ foo }}</div>
</div>
</template>

<style scoped>
.foo {
  color: red;
}
</style>"
`;

exports[`wrap html 1`] = `
"<template>
  <foo><h1>hello</h1>
</foo>
</template>

"
`;
